client/src/components/dashboard.js

```diff
@@ -38,26 +42,24 @@ class Dashboard extends Component {
         }
 
         await web3api.initWeb3Connection();
-        this.grantContract = contract(GrantContractArtifact);
-        this.grantContract.setProvider(web3api.web3.currentProvider);
 
-        this.setState({projects: await this.fetchProjects(),
-                       spinning: false});
-
-
-        // Listening to the "projectAdded" event
+        let grantFacrotyContract = contract(GitFundedGrantFactory);
+        grantFacrotyContract.setProvider(web3api.web3.currentProvider);
         try {
-            let grantContractInstance = await this.grantContract.deployed();
-            grantContractInstance.projectAdded((error, result) => {
+            this.grantFacrotyContract = await grantFacrotyContract.deployed();
+
+            // Listening to the "projectAdded" event
+            this.grantFacrotyContract.projectAdded((error, result) => {
                 this.setState({projects: this.state.projects.concat(result.args)})
             });
         }
-        catch(err) {
-            console.log(err);
-        }
-
+        catch(error) {
+            this.setState({networkError: true})
 
+        }
 
+        this.setState({projects: await this.fetchProjects(),
```

Comment:
```
const projects = await this.fetchProjects;
this.setState({projects});
```

Reply:
Ignoring this as there are multiple states to set.